modules/common/pod/pod.go

```diff
@@ -47,3 +47,21 @@ func GetPodListWithLabel(
 
 	return podList, nil
 }
+
+// GetHeadlessPodFQDN gets a list of pods matching the labels provided and returns a slice of pod FQDNs. This
+// is required in order to allow resolution of pods behind Headless services.
+func GetHeadlessPodFQDN(ctx context.Context, h *helper.Helper, serviceName string, namespace string, labelSelector map[string]string) ([]string, error) {
+	var podSvcNames []string
+	var podList *corev1.PodList
+
+	podList, err := GetPodListWithLabel(ctx, h, namespace, labelSelector)
+	if err != nil {
+		return []string{}, fmt.Errorf("Error getting list of pods: %v", err)
+	}
+
+	for _, pod := range podList.Items {
+		podSvcNames = append(podSvcNames, fmt.Sprintf("%s.%s", pod.Name, serviceName))
```

Comment:
I think we could make this more generic and don't have to pass in the service name, just namespace and a  pod label selector. then check if `spec.hostname` and `spec.subdomain` is set properly on the pod.

Reply:
and it would remove the limitation on heatless pods

Reply:
Ah sorry, you have asked me to do that a couple of times. My bad, latest update implements this.